server/channels/app/webhook.go

```diff
@@ -831,16 +833,16 @@ func (a *App) HandleIncomingWebhook(c request.CTX, hookID string, req *model.Inc
 	}
 
 	if hook.ChannelLocked && hook.ChannelId != channel.Id {
-		return model.NewAppError("HandleIncomingWebhook", "web.incoming_webhook.channel_locked.app_error", nil, "", http.StatusForbidden)
+		return model.NewAppError("HandleIncomingWebhook", "web.incoming_webhook.channel_locked.app_error", map[string]any{"channel_id": channel.Id}, "", http.StatusForbidden)
 	}
 
 	resultU := <-uchan
 	if resultU.NErr != nil {
-		return model.NewAppError("HandleIncomingWebhook", "web.incoming_webhook.user.app_error", nil, "", http.StatusForbidden).Wrap(resultU.NErr)
+		return model.NewAppError("HandleIncomingWebhook", "web.incoming_webhook.user.app_error", map[string]any{"user": hook.UserId}, "", http.StatusForbidden).Wrap(resultU.NErr)
```

Comment:
Should we use `user_id` here instead of `user` as the key?

Reply:
This one is a special case, where the same error "can't find the user" is used at 2 search attempts, one by user_id and another by username.
So, went with the `user` key.

Reply:
I don't think the property name should match the error message, but describe the field being logged. Is this always guaranteed to be a `user_id`?

Reply:
(Consider the typical use case of scrubbing logs with Loki, and filtering on `user_id=XXX`. It's ideal to have this find all related results, and not have to remember to do `user=XXX` for cases like this.)

Reply:
Actually, my intention wasn’t to match the error message exactly but rather to account for both cases within a single error message. 
In this implementation, it’s not always guaranteed to be `user_id`; in one case, we’re passing `username`, and in another, `user_id` to the _same_ error message.

For the log filtering, it would be even better to standardize with a generic key like `user`. This way, whether the search term is `user_id` or `username`, we could simply filter by consistently using `user=xxx`.
Also, I'm not sure how the Loki search works, but couldn't you simply search for `xxx` or `=xxx` no matter the key?

Reply:
> For the log filtering, it would be even better to standardize with a generic key like user.

I think the momentum is already behind naming the keys explicitly. But in this case, if it's sometimes both, then it's probably fine :)

Reply:
> Also, I'm not sure how the Loki search works, but couldn't you simply search for xxx or =xxx no matter the key?

That is not idiomatic way to use Loki, although yes, it's doable. The idiomatic way is to use structured logging, and filter with the keys. This is much more powerful, and you can use Loki in-built utilities to filter for the value of a given `user_id` key, do aggregation operations, and so on.

This is unrelated to this specific case where the same error is being used for username and user_id both. But just a general idea on idiomatic logging for better usage of Loki.